### PR TITLE
Sunrise integration

### PR DESCRIPTION
**Pull request by fenric**
Please look to `scripts/test-all-factory`. Sunrise does not support composer "--prefer-lowest" mode due to the fact that in the first versions there was no factory...

### CODE REVIEW

File: scripts/test-all-factory [hunk 1]
```diff
@@ -25,6 +25,14 @@ testImplementation http-interop/http-factory-guzzle "^1.0" || exit 1
 testImplementation http-interop/http-factory-slim "^1.0" || exit 1
 testImplementation zendframework/zend-diactoros "^2.0" || exit 1
 
+if [[ "$COMPOSER_FLAGS" == *"--prefer-lowest"* ]]
+then
+    echo "Sunrise package missed due to unsupported composer \"--prefer-lowest\" mode"
+    echo
+else
+    testImplementation sunrise/http-factory "^1.0" || exit 1
```
Comment (shadowhand) [inline scripts/test-all-factory]: Use the lowest version that contains the factory implementation so that this workaround is not necessary.

**Review by shadowhand — CHANGES_REQUESTED** [review]

File: scripts/test-all-factory (hunk 1 above)
Comment (fenric) [reply scripts/test-all-factory]: The repository [http-factory](https://github.com/sunrise-php/http-factory) does not contain factories, it pulls other repositories:
[http-message](https://github.com/sunrise-php/http-message), [http-server-request](https://github.com/sunrise-php/http-server-request), [stream](https://github.com/sunrise-php/stream), [uri](https://github.com/sunrise-php/uri)

It's problem... What better way to do?

Comment (shadowhand) [reply scripts/test-all-factory]: Correct the minimum version requirement in `http-factory` to point to a `stream` implementation that contains the factory.

Comment (fenric) [reply scripts/test-all-factory]: Good solution, thank you very much, all okay?

**Review by shadowhand — APPROVED** [review]

**Comment by shadowhand** [comment]
Thank you! Released in version 1.5.0.

**Pull request merged**